doc/src/modules/physics/mechanics/kane.rst

```diff
@@ -22,12 +22,37 @@ differentiated non-holonomic equations.
   \mathbf{k_d}(q, t) \dot{u} + \mathbf{f_d}(q, \dot{q}, u, t) &= 0\\
   \mathbf{k_{dnh}}(q, t) \dot{u} + \mathbf{f_{dnh}}(q, \dot{q}, u, t) &= 0\\
 
-In :mod:`mechanics` holonomic constraints are only used for the linearization
-process; it is assumed that they will be too complicated to solve for the
-dependent coordinate(s).  If you are able to easily solve a holonomic
-constraint, you should consider redefining your problem in terms of a smaller
-set of coordinates. Alternatively, the time-differentiated holonomic
-constraints can be supplied.
+The first equation is non-linear in the generalized coordinates and typically
+describes configuration constraints like kinematic loops. For the purposes of
+forming the non-linear equations of motion, these are additional equations that
+must be satisfied along with the other equations. In :mod:`mechanics` holonomic
+constraints are only used for the linearization process; it is assumed that
+they will be too complicated to solve for the dependent coordinate(s). If you
+are able to easily solve a holonomic constraint, you should consider redefining
+your problem in terms of a smaller set of coordinates. Alternatively, the
+time-differentiated holonomic constraints can be supplied.
+
+The second equation describes the non-holonomic constraints, i.e. velocity
+constraints, that are linear in the generalized speeds. There are fewer
+equations than generalized speeds, and thus describe the relationship between
+the dependent and independent generalized speeds. Where there are :math:`m`
+dependent speeds and :math:`o-m` independent speed.
```

Comment:
Yes, I'll define all of these above:

- o: number of generalized speeds
- m: number of non-holonomic constraints
- n:
- p:

The last two are related to number of indpendent speeds I think. We are using Kane's notation for that, I'll look in his book.